### PR TITLE
New bug form

### PR DESCRIPTION
**Pull request by Felonious-Spellfire**
## What does this PR do?

Created issue form using BETA YAML file support and removes MD issue form.

### Does your PR introduce any breaking changes? If yes, please list them.

I'm not sure of the impact on this: https://github.com/Lightning-AI/lightning/pull/14017/files

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [X] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+name: Report a bug
+description: Errors and regression reports with complete reproducing test cases.
+labels: ["status:triage", "bug"]
+body:
+  - type: markdown
+    attributes:
+      value: >
+        Bug reports aren't usually really **🐛 bugs**. More often than not they are questions about how to use the product.
+        Hit us up on our [Community Slack](https://pytorch-lightning.slack.com) first if that is the case.
+        You're probably going to get a faster response and you're going to learn about using the product.
```
Comment (rohitgr7) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:10]:
```suggestion
        Bug reports aren't usually really **🐛 bugs**. More often than not, they are questions about how to use the product.
        If that is the case, hit us up on our [Community Slack](https://pytorch-lightning.slack.com) first.
        You're probably going to get a faster response, and you're going to learn about using the product.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+name: Report a bug
+description: Errors and regression reports with complete reproducing test cases.
+labels: ["status:triage", "bug"]
+body:
+  - type: markdown
+    attributes:
+      value: >
+        Bug reports aren't usually really **🐛 bugs**. More often than not they are questions about how to use the product.
+        Hit us up on our [Community Slack](https://pytorch-lightning.slack.com) first if that is the case.
```
Comment (rohitgr7) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:9]: I'd suggest GH discussions. Will reduce the prob of repeated questions we get.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        1. A clear and concise description of the bug - keep it to a line or two
+
+        2. A clear and concise code example that reproduces the problem. To get that use the [BoringModel] (https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb)!
+
+         IMPORTANT: Has to be public.
+
+         If you could not reproduce using the BoringModel and still think there's a bug, please post here
+         but remember, bugs with code are fixed faster!
+
+        3. Additional details that may help us reproduce your issue.
```
Comment (rohitgr7) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:26]:
```suggestion
        If you are new to making bug reports, here are a few guidelines for you:

        1. A clear and concise description of the bug - keep it to a line or two

        2. A clear and concise code example that reproduces the problem. To get that, use the [BoringModel] (https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb)!

         IMPORTANT: It has to be public.

         If you could not reproduce using the BoringModel and still think there's a bug, please post it here
         but remember, bugs with code are fixed faster!

        3. Additional details that may help us reproduce your issue.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      label: Bug summary
+      description: A clear and concise description of the 🐛bug🐛.
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Reproduction
+      description: >
+        Provide your clear and concise code example that reproduces the problem. To get that use the [BoringModel] (https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb)
```
Comment (rohitgr7) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:52]:
```suggestion
        Provide your clear and concise code example that reproduces the problem. To get that, use the [BoringModel] (https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb)
```

**Review by rohitgr7 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (justusschock) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:9]: GH discussions is definitely better in terms of indexing for others but worse in getting notifications for us to answer.

Comment (rohitgr7) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:9]: we could add a GH notification app on a slack channel for discussions.
but looks like it's not there yet: https://github.com/integrations/slack/issues/1223, https://github.com/integrations/slack/issues/1134

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
+name: Report a bug
+description: Errors and regression reports with complete reproducing test cases.
+labels: ["status:triage", "bug"]
```
Comment (akihironitta) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:3]: Ignore my comment here if we're changing the current label "needs triage".
```suggestion
labels: ["needs triage", "bug"]
```

**Review by akihironitta — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (edenlightning) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:9]: Is it
```suggestion
        Do you have a question? Look it up in GitHub discussion. Can't find an answer? Create a new issue, or hit us up on our [Community Slack](https://pytorch-lightning.slack.com).
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        You're probably going to get a faster response and you're going to learn about using the product.
+
+  - type: markdown
+    attributes:
+      value: >
+        If you are new to making bug reports, here are a few guidelines for you:
+
+        1. A clear and concise description of the bug - keep it to a line or two
+
+        2. A clear and concise code example that reproduces the problem. To get that use the [BoringModel] (https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb)!
```
Comment (edenlightning) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:19]: Boring model applies only to trainer bugs. What should we do for apps? @manskx @hhsecond

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    attributes:
+      value: >
+        Bug reports aren't usually really **🐛 bugs**. More often than not they are questions about how to use the product.
+        Hit us up on our [Community Slack](https://pytorch-lightning.slack.com) first if that is the case.
+        You're probably going to get a faster response and you're going to learn about using the product.
+
+  - type: markdown
+    attributes:
+      value: >
+        If you are new to making bug reports, here are a few guidelines for you:
```
Comment (edenlightning) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:15]: This is a bit repertative? They can already learn this in the structured fields

**Review by edenlightning — COMMENTED** [review]
This is great! Can we duplicate this, and have another template for Lightning.ai UI issues?

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 7 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Removed

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Modified

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 5 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Modified

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 3 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Removed this

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Modified this

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 1 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Modified

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 4 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Removed this

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 6 above)
Comment (Felonious-Spellfire) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Removed this

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (rohitgr7) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: I can't seem to find the discussion link, Still slack link is attached

**Comment by awaelchli** [comment]
We had good instructions how the user can collect the environment information, with a simple command and script we provide. This is now gone. 

This is valuable information that is now completely missed. The result is: 

- More work for us: If the information is not provided, we have to ask and tell them how to collect it. This increases the time it takes for us to arrive at a reproducible problem, and delays the arrival of a fix. 
-  More work for them: Instead of getting simple command to run, then copy paste the information into the issue, the user now has to collect this manually and type it into the issue page. 

Can we consider adding this back? This is very important to me, and I believe also for the team. 

cc @edenlightning @Borda @carmocca

**Comment by Felonious-Spellfire** [comment]
@awaelchli @edenlightning @Borda @carmocca 

https://github.com/Lightning-AI/lightning/pull/14736

File: .github/ISSUE_TEMPLATE/1_bug_report.yaml [hunk 8]
```diff
@@ -0,0 +1,98 @@
+name: Report a bug
+description: Any errors that are not UI related.
+labels: ["needs triage", "bug"]
```
Comment (carmocca) [inline .github/ISSUE_TEMPLATE/1_bug_report.yaml:3]: @Felonious-Spellfire This should be kept as just "needs triage" as it was in the previous file. Often users misclassify as bugs, and all issues need to be triaged anyways by a maintainer.

**Pull request merged**